### PR TITLE
User Can Update Existing Food

### PR DESCRIPTION
**Pull request by alexander-mathieu**
This PR adds an endpoint for a user to update an existing food in the database.

- Adds spec user_can_update_existing_food to foods_api_endpoint spec
- Adds food update route to foodsRouter

### CODE REVIEW

File: tests/api/v1/foods_api_endpoint.spec.js [hunk 1]
```diff
@@ -72,19 +77,62 @@ describe('foods api endpoint', () => {
...
+        expect(Object.keys(response.body)).toContain('name')
+        expect(Object.keys(response.body)).toContain('calories')
+
+        expect(Object.keys(response.body)).not.toContain('createdAt')
+        expect(Object.keys(response.body)).not.toContain('updatedAt')
+      })
+    })
+  })
+
+  test('user recieves a 404 error when no food is found to update', () => {
```
Comment (CosmicSpagetti) [inline tests/api/v1/foods_api_endpoint.spec.js]: nice Testing edge cases
